packages/attestation-service/src/requestHandlers/attestation.ts

```diff
@@ -222,31 +245,28 @@ class AttestationRequestHandler {
         this.identifier,
         transaction
       )
-      const provider = smsProviderFor(this.attestationRequest.phoneNumber)
-
-      if (!provider) {
-        await attestationRecord.update(
-          { status: AttestationStatus.UNABLE_TO_SERVE, smsProvider: SmsProviderType.UNKNOWN },
-          { transaction, logging: this.sequelizeLogger }
-        )
-        await transaction.commit()
-        Counters.attestationRequestsUnableToServe.inc()
-        return attestationRecord
-      }
 
       try {
-        await provider.sendSms(this.attestationRequest.phoneNumber, textMessage)
-        this.logger.info('Sent sms')
+        const sentVia = await startSendSms(
+          this.attestationRequest.phoneNumber,
+          textMessage,
+          () => {
+            void this.persistFinalAttesationResult(AttestationStatus.FAILED)
+          },
+          () => {
+            void this.persistFinalAttesationResult(AttestationStatus.COMPLETE)
+          }
+        )
         Counters.attestationRequestsSentSms.inc()
         await attestationRecord.update(
-          { status: AttestationStatus.SENT, smsProvider: provider.type },
+          { status: AttestationStatus.SENT, smsProvider: sentVia },
```

Comment:
it looks like this `AttestationStatus.SENT` would overwrite the persisted `AttestationStatus.FAILED` or `AttestationStatus.COMPLETE` from the callbacks above.

Reply:
I don't think this could happen because the record is locked and so the `FAILED` or `COMPLETE` could never get written before the `SENT` does?   But I don't love recording this stuff to the database that much. It's not really used for anything other than preventing replays that cause you to spend $$ sending the same SMS over and over -- yet we store a bunch of status in there too.

Reply:
but we are doing `await startSendSms()` before we update to `SENT`.  And `startSendSms()` doesn't return without triggering one of these callbacks.  

so I don't see how they wouldn't be set to `COMPLETE`/`FAILURE` and then overwritten as `SENT`

Reply:
because the callbacks need to wait to acquire the lock on the row

Reply:
oh i see -- earlier in this method we already acquire the lock.  my understanding is: if there is another attempt to update the row like the `COMPLETE`/`FAILURE` callbacks, it would just fail rather than wait for the unlock.  and even if it waited, then you'd be stuck waiting at `await startSendSms()` before being able to run ` await attestationRecord.update(SENT)`

the sequelize [documentation](https://sequelize.org/master/class/lib/transaction.js~Transaction.html) isn't explicit about this, but i've seen the error case with locked rows in the other lightweight ORM used for PGPNP.  

what are you seeing in practice?

Reply:
removed for now per discussion